### PR TITLE
[Infer] Fix transformers streaming logprobs

### PR DESCRIPTION
**Pull request by taking-lying-flat**
## Summary

- retain the `LogitsStreamer` instance used by PT/Transformers generation so streaming responses can consume its logits

## Root cause

The PT streaming path constructed `LogitsStreamer()` inline inside `LogitsProcessorList`, while the local `logits_streamer` variable remained `None`. The collection path therefore returned early for every streamed chunk, and `logprobs` was serialized as `null`.

## Impact

Requests using the Transformers backend with both `stream=true` and `logprobs=true` now return populated logprobs. Non-streaming inference and streaming requests without logprobs are unchanged.

## Validation

- `python -m py_compile swift/infer_engine/transformers_engine.py`
- `git diff --check upstream/main...HEAD`

### CODE REVIEW

**Review by hjh0119 — APPROVED** [review]

**Comment by hjh0119** [comment]
thanks
